### ISSUE TITLE
Default reviewers not registering on Create Pull Request form

### ISSUE THREAD

**Issue by dafelipe130** [monitorjbl/pr-harmony#24]
Default reviewers are not registering on the reviewers field on creation of a pull request regardless if they're entered into the PR Harmony repository configuration.

They show up though if they are configured on a project instead of a repository. However, our use case would be better served if the repository configuration is honored regarding pull request options (branch protection options works fine though)

PR Harmony 1.8
Bitbucket Server (Stash) 3.11.2

**Comment by monitorjbl** [comment]
Hm, that is odd...the way this works on the backend, there really can't be a one-or-the-other kind of bug. Are you positive that the repo configuration saved properly? Are you able to see the changes when you go back to the repo config screen?

**Comment by dafelipe130** [comment]
Other than the configuration page refreshing upon clicking "Save", I'm assuming the configuration is pushing though. Also, the changes are indeed present when going back to the screen.

**Comment by monitorjbl** [comment]
Can you confirm on a repo that has not had any project-level configuration that the backend endpoint is giving the correct output then? That's what the PR create screen depends on to do the Javascript hook in the UI. You can access it by going to 

```
http://<base-url>/rest/pr-harmony/1.0/config/<project-key>/<repo-key>
```

For example, if I was running this locally, my base URL would be `http://localhost:7990/stash` so the complete URL for a repo would be `http://localhost:7990/bitbucket/rest/pr-harmony/1.0/config/PROJECT_1/rep_1`

**Comment by dafelipe130** [comment]
Thanks for responding to this issue. Your help is very much appreciated.

Just checked the backend endpoint of a repository without any project configuration and it seems to be giving me the correct output.

> {"defaultReviewers":["amarsh"],"defaultReviewerGroups":[],"requiredReviewers":[],"requiredReviewerGroups":[],"blockedCommits":["master"],"blockedPRs":[],"automergePRs":[],"automergePRsFrom":[],"excludedUsers":["vgroux","bamboo"],"excludedGroups":[]}

I also noted that the our service runs behind an SSL reverse proxy and the certificate for this has already expired. Could it be that I'm facing authentication issues when configuring pull request options on a repository level?

**Comment by monitorjbl** [comment]
I wouldn't necessarily rule that out, but it's more likely the Javascript hook has a bug in it. Do you know how to use the Javascript console in your browser? If so, check it while you try to open a PR. I suspect it's throwing an error.

**Comment by dafelipe130** [comment]
I'm getting a 

> JQMIGRATE: jQuery.parseJSON requires a valid JSON string

warning. 

Other than that, I got a slew of deprecation warnings. Should I post the full Console log and screenshots?

**Comment by monitorjbl** [comment]
That would be awesome. I know Stash throws a ton of stuff to that log and most of it is irrelevant.

**Comment by dafelipe130** [comment]
![log](http://i.imgur.com/xp8FYxD.png)

JQMIGRATE: Logging is active
JQMIGRATE: jQuery.browser is deprecated
console.trace()
DEPRECATED - requiring 'eve' has been deprecated since AUI 5.7 and will be removed in AUI 6.0.
DEPRECATED JS - Inline dialog constructor has been deprecated and will be removed in a future release. Use inline dialog 2 instead.  
     at Function.InlineDialog (https://stash.valuex.com/s/e58528390262dbcc787930cd623631be-CDN/en_US/0329d81/15/15/_/download/superbatch/js/batch.js?locale=en-US:229:506)
DEPRECATED JS - Dialog constructor has been deprecated and will be removed in a future release. Use Dialog2 instead.  
     at null.<anonymous> (https://stash.valuex.com/s/e58528390262dbcc787930cd623631be-CDN/en_US/0329d81/15/15/_/download/superbatch/js/batch.js?locale=en-US:229:506)
DEPRECATED JS - Dialog popup constructor has been deprecated and will be removed in a future release.  
     at Function.popup (https://stash.valuex.com/s/e58528390262dbcc787930cd623631be-CDN/en_US/0329d81/15/15/_/download/superbatch/js/batch.js?locale=en-US:229:506)
Fetching required users
Loaded required users
JQMIGRATE: jQuery.parseJSON requires a valid JSON string
console.trace()
DEPRECATED undefined - DisplayAbove has been deprecated and will be removed in a future release.  
     at Object.g.reset (https://stash.valuex.com/s/e58528390262dbcc787930cd623631be-CDN/en_US/0329d81/15/15/_/download/superbatch/js/batch.js?locale=en-US:845:343)
Fetching required users
Loaded required users
Deprecation warning: moment().zone is deprecated, use moment().utcOffset instead. https://github.com/moment/moment/issues/1779
Deprecation warning: moment().add(period, number) is deprecated. Please use moment().add(number, period).
Deprecation warning: moment().subtract(period, number) is deprecated. Please use moment().subtract(number, period).
Fetching required users
Fetching required users
Loaded required users

**Comment by monitorjbl** [comment]
Based on the logging output you pasted, it certainly looks like its doing everything right. Are you logged as that "amarsh" user? Stash won't let you add yourself to the PR, so that might explain what's going on.

**Comment by dafelipe130** [comment]
> Are you logged as that "amarsh" user?
> Nope, that's a different

Tried going for the JSON output again and I seem to have gotten the correct output now. (no values regardless of whats in the configuration page)

> {"defaultReviewers":[],"defaultReviewerGroups":[],"requiredReviewers":[],"requiredReviewerGroups":[],"blockedCommits":[],"blockedPRs":[],"automergePRs":[],"automergePRsFrom":[],"excludedUsers":[],"excludedGroups":[]}

![JSON](http://i.imgur.com/Iac6FJi.png)

Sorry about the earlier post. Must have gotten a cached page

**Comment by cooksimo** [comment]
I'm having a similar issue and have tracked it down to a case sensitivity problem.
We have a repo named iOSApp, the config page uses this REST call:
https://stash/rest/pr-harmony/1.0/config/IR/iosapp

The create PR page uses this one:
https://stash/rest/pr-harmony/1.0/users/IR/iOSApp

Seems these url's are case sensitive and hence return different results.

**Comment by cooksimo** [comment]
This is a guess, but should line 121 in https://github.com/monitorjbl/pr-harmony/blob/master/src/main/resources/pr-create-hook.js#L121 read:
`url: root + '/rest/pr-harmony/1.0/users/' + target.project.key + '/' + target.slug,`

**Comment by monitorjbl** [comment]
Interesting...I'll take a look at your proposed solution and see if I can get it to work locally.

**Comment by monitorjbl** [comment]
Yep, confirmed. Thanks for the detective work, I don't think that would ever had occurred to me! All of our repos are lowercased.

**Comment by dafelipe130** [comment]
Hello,

Can confirm on my case as well. Almost all our repositories have at least one uppercase letter in their names. Would also like to add that the problem also surfaces on repositories having spaces in their names. For e.g. "my repository" would not work but "my-repository" will.

**Comment by monitorjbl** [comment]
@dafelipe130 this was fixed in release 2.1 and 1.8.1 for Bitbucket and Stash,  respectively.

**Comment by dafelipe130** [comment]
Hello,

Have updated our Stash instance with the fixed version of PR Harmony and it now works for our specific use case.

Thanks everyone for contribution. You may close this ticket :)
